R/tm_t_crosstable.R

```diff
@@ -150,18 +150,17 @@ ui_t_crosstable <- function(id, datasets, x, y, show_percentage, show_total, pre
         )
       )
     ),
-    forms = teal::get_rcode_ui(ns("rcode")),
+    forms = teal.widgets::verbatim_popup_ui(ns("rcode"), "Show R code"),
     pre_output = pre_output,
     post_output = post_output
   )
 }
 
-srv_t_crosstable <- function(id, datasets, reporter, label, x, y, basic_table_args) {
+srv_t_crosstable <- function(id, data, reporter, filter_panel_api, label, x, y, basic_table_args) {
```

Comment:
```
library(teal.modules.general)

x <- data.frame(id = 1:10, pk = c(1:5, 1:5), val = 1:10)
y <- data.frame(id = 1:5, value = 1:5)

app <- init(
  data = teal_data(
    dataset("X", x),
    dataset("Y", y),
    join_keys = join_keys(join_key("X", "Y", c("pk" = "id")))
  ),
  modules = modules(
    tm_t_crosstable(
      label = "Cross Table",
      x = data_extract_spec(
        dataname = "X",
        select = select_spec(
          label = "Select variable:",
          choices = variable_choices(x, "val"),
          selected = "val",
          multiple = TRUE,
          ordered = TRUE,
          fixed = FALSE
        )
      ),
      y = data_extract_spec(
        dataname = "Y",
        select = select_spec(
          label = "Select variable:",
          choices = variable_choices(y, "value"),
          selected = "value",
          multiple = FALSE,
          fixed = FALSE
        )
      )
    )
  )
)

shinyApp(app$ui, app$server)

```

So if I select a different join, the SRC (and maybe the output?) still says full_join

![image](https://user-images.githubusercontent.com/15201933/187476057-d5eca1e2-2629-4e58-83de-cc858a0683e3.png)

Reply:
merge_fun is used instead of join_fun

I will fix the branch and create a PR for main

Reply:
https://github.com/insightsengineering/teal.modules.general/pull/452